blocks/blockstore/write_cache.go

```diff
@@ -8,7 +8,7 @@ import (
 )
 
 // WriteCached returns a blockstore that caches up to |size| unique writes (bs.Put).
-func WriteCached(bs Blockstore, size int) (Blockstore, error) {
+func WriteCached(bs Blockstore, size int) (*writecache, error) {
```

Comment:
what i mean is getting into an annoying case where i need the real type of the thing but i don't have it -- maybe we should also export `WriteCache`? **shrug** it's fine i guess